service/storage_proxy.cc

```diff
@@ -3037,7 +3059,8 @@ storage_proxy::create_write_response_handler_helper(schema_ptr s, const dht::tok
 
     auto all = boost::range::join(natural_endpoints, pending_endpoints);
 
-    if (cannot_hint(all, type)) {
+    // TODO: Use small_vector here instead to avoid dynamic allocations.
+    if (cannot_hint(to_host_id(all, get_token_metadata_ptr()), type)) {
```

Comment:
we should use `erm->get_token_metadata()` since this is what we used to construct natural_endpoints and pending endpoints